ashura/engine/tests/animation_test.cc

```diff
@@ -0,0 +1,543 @@
+  }
+
+  // Calculate table dimensions including borders
+  std::pair<int, int> get_table_dimensions(const TerminalTable &table)
+  {
+    std::stringstream ss(table.to_string());
+    std::string       line;
+    int               width = 0, height = 0;
+
+    while (std::getline(ss, line))
+    {
+      width = std::max(width, static_cast<int>(line.length()));
+      height++;
+    }
+    return {width, height};
+  }
+
+  void validate_id(const std::string &id) const
+  {
+    if (id.empty())
+    {
+      throw std::invalid_argument("Table ID cannot be empty");
+    }
+  }
+
+  void validate_position(const Position &pos) const
+  {
+    if (pos.x < 0 || pos.y < 0 || pos.x >= terminal_width ||
+        pos.y >= terminal_height)
+    {
+      throw std::out_of_range("Position is outside terminal bounds");
+    }
+  }
+
+  void initialize_screen()
+  {
+    screen = std::vector<std::vector<char>>(
+        terminal_height, std::vector<char>(terminal_width, ' '));
+  }
+
+  void write_to_screen(int startY, int startX, const std::string &text)
+  {
+    if (startY >= 0 && startY < terminal_height)
+    {
+      size_t xPos = static_cast<size_t>(startX);
+      for (char c : text)
+      {
+        if (xPos < static_cast<size_t>(terminal_width))
+        {
+          screen[startY][xPos++] = c;
+        }
+      }
+    }
+  }
+
+  int get_table_height(const TableInfo &table_info) const
+  {
+    std::stringstream ss(table_info.table->to_string());
+    int               height = 0;
+    std::string       line;
+    while (std::getline(ss, line))
+    {
+      height++;
+    }
+    return height + (table_info.title.empty() ?
+                         0 :
+                         1);        // Add 1 for title if present
+  }
+
+  void clear_screen_area(int startY, int endY)
+  {
+    for (int y = startY; y <= endY && y < terminal_height; ++y)
+    {
+      for (int x = 0; x < terminal_width; ++x)
+      {
+        screen[y][x] = ' ';
+      }
+    }
+  }
+
+public:
+  TableManager(int width = 80, int height = 24) :
+      terminal_width(width), terminal_height(height)
+  {
+    if (width <= 0 || height <= 0)
+    {
+      throw std::invalid_argument("Invalid terminal dimensions");
+    }
+    initialize_screen();
+  }
+
+  // Add a new table with position and identifier
+  void add_table(const std::string &id, std::unique_ptr<TerminalTable> table,
+                 Position pos, const std::string &title = "")
+  {
+    validate_id(id);
+    validate_position(pos);
+    tables.try_emplace(id, TableInfo(std::move(table), pos, title));
+  }
+
+  // Get reference to a table for updating
+  TerminalTable &get_table(const std::string &id)
+  {
+    auto it = tables.find(id);
+    if (it == tables.end())
+    {
+      throw std::runtime_error("Table not found: " + id);
+    }
+    return *(it->second.table);
+  }
+
+  void set_table_position(const std::string &id, Position new_pos)
+  {
+    auto it = tables.find(id);
+    if (it != tables.end())
+    {
```

Comment:
I'll be removing all the code in here as it isn't really a good test of the animation system. we use unit testing where reasonable, but for this case, I think we're okay without tests.

Reply:
also, you should try to keep tests scope small, I don't think it's a good idea to integrate a new TUI to a test suite.